Sources/Controllers/Map/Layers/OAMyPositionLayer.mm

```diff
@@ -603,6 +608,10 @@ - (BOOL) updateLayer
         _textScaleFactor = textScaleFactor;
         [self refreshMarkersCollection];
     }
+    if (_lastMarkersUpdatedMode != [OAAppSettings sharedManager].applicationMode.get)
+    {
+        [self refreshMarkersCollection];
```

Comment:
Not clear why we should refresh all markers if app mode switched. It even not sign that any of app's mode setting changed.
Probably we should have some observer like `onAppModeSettingsChanged` and then recreate only this maker from collection.